### ISSUE TITLE
Commands and prediction should consistently update world state after being applied

### ISSUE THREAD

**Issue by SomeWritesReserved** [SomeWritesReserved/Entmoot#4]
Currently **CommandData.ApplyToEntity** assumes that it will update the entity right then and there but this circumvents the whole "systems perform work" paradigm. This is problematic because the server might get multiple commands in the same frame and apply them but this would have different net output than if they were received one frame at a time (because collision would happens in between frames while applying commands in one frame means no collisions, only collisions on the end result); this is how prediction works (command data applied then prediction occurs after each application).

Example: frame 1 move forward, frame 2 move right. Right now this is a net movement of (1,1) if they were both received in the same frame, which could clip a corner. If updates were after each command application then it would be (1,0) then (0,1) which would clear a corner.

I think ideally (though I haven't thought this through) is that command data would modify components and then all systems get updated but only for the commanding entity, then the next command data is applied to components, rinse repeat.

**Comment by SomeWritesReserved** [comment]
See commit 12c249e98199405c30577deb2602653d3af0c5e0 for an attempt at this and a commit message as to what failed.

**Comment by SomeWritesReserved** [comment]
Previous information from the wiki TODO list:

```
* Server-side: multiple client commands are applied at once and then updated a single time.
  * Instead the entity could be updated after each command is applied. This is how prediction works. This is important because otherwise getting two commands at once behaves differently from getting two commands over two frames.
    * Example: frame 1 move forward, frame 2 move right. Right now this is a net movement of (1,1) which could clip a corner. If updates were after each command application then it would be (1,0) then (0,1) which would clear a corner.
  * Or CommandData.ApplyToEntity could just update the entity right then and there but would need access to the other systems and the EntityArray.
    * This would replace the need for a dummy component to hold player data and a player movement system and also the ClientPrediction method on systems altogether.
    * See https://github.com/SomeWritesReserved/Entmoot/commit/12c249e98199405c30577deb2602653d3af0c5e0
```

**Comment by SomeWritesReserved** [comment]
Apparently this was on my brain way back in Oct 2017, see f205a6c799063b8cee334abb7bef4181b554f021. Ultimately I was concerned for different reasons but came up with essentially the same solution as proposed above (have systems take care of command processing).

**Comment by SomeWritesReserved** [comment]
Fixed under ca30472e0651ed5da04b1036f3c2b9dd2cafd259. See the commit message for extensive details.
